WordPress/gutenberg
# Site Editor: Revert custom templates to their original theme-provided files

## Description

Fixes #23421

- Create a new `core/edit-site` action that reverts a template to its original theme-provided file.
- Add a new "Revert" button in the Site Editor's document action for revertable templates.

### Technical Notes

"Reverting a template" means:
- Delete the current template.
- Let the editor fetch a new version of the current template, which at this point should be the original file version.

"Revertable template" indicates a template that `source === 'custom'` and `original_file_exists` (new property).

## How has this been tested?

- Activate an FSE theme such as TT1-Blocks.
- Open the Site Editor and customize the front page template (make sure you are modifying the actual template, not one of its template part blocks).
- Save: this will create a custom version of the front page template, stored as `wp_template` CPT.
- Open the document actions (in the editor header), and make sure it now shows a "Revert" button.
- Click it!
- Save
- Make sure the custom template has been trashed (you can find it in `wp-admin/edit.php?post_status=trash&post_type=wp_template`).
- Make sure the site editor shows the original template content

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
